linera-execution/src/execution_state_actor.rs

```diff
@@ -336,8 +336,24 @@ where
                     .map(|http::Header { name, value }| Ok((name.parse()?, value.try_into()?)))
                     .collect::<Result<HeaderMap, ExecutionError>>()?;
 
+                let url = Url::parse(&request.url)?;
+                let host = url
+                    .host_str()
+                    .ok_or_else(|| ExecutionError::UnauthorizedHttpRequest(url.clone()))?;
+
+                let (_epoch, committee) = self
+                    .system
+                    .current_committee()
+                    .ok_or_else(|| ExecutionError::UnauthorizedHttpRequest(url.clone()))?;
+                let allowed_hosts = &committee.policy().http_request_allow_list;
+
+                ensure!(
+                    allowed_hosts.contains(host),
+                    ExecutionError::UnauthorizedHttpRequest(url)
+                );
+
                 let response = Client::new()
-                    .request(request.method.into(), request.url)
+                    .request(request.method.into(), url)
```

Comment:
Should we whitelist also methods?

Reply:
I don't think that's needed. We could make the allow list more complicated, mapping hosts to a set of allowed methods, but where do we draw the line? We could also limit headers, ports, etc. Right now I'm assuming that the allowed HTTP servers will be safe to communicated with.